### PR TITLE
Transform relationship labels

### PR DESCRIPTION
**Pull request by bengolder**
- use a standard gender neutral label for all pdf filling
- for NY XML, use the direct file key without spaces
- for AZ XML, use a specific lookup table to translate

### CODE REVIEW

File: app/lib/pdf_filler/az140_pdf.rb
Comment (bengolder) [inline app/lib/pdf_filler/az140_pdf.rb:1]: For the PDF, I needed to pull dependents from the database instead of the XML so that I could use the `relationship_label` method. For selecting which dependents to iterate over in each section, I copied the logic from the XML for consistency.

File: app/lib/submission_builder/ty2022/states/az/individual_return.rb [hunk 1]
```diff
@@ -4,12 +4,15 @@ module Ty2022
     module States
       module Az
         class IndividualReturn < SubmissionBuilder::Document
-          FILING_STATUSES = {
-            single: 'Single',
-            married_filing_jointly: 'MarriedJoint',
-            married_filing_separately: 'MarriedFilingSeparateReturn',
-            head_of_household: 'HeadHousehold',
-          }.freeze
```
Comment (bengolder) [inline app/lib/submission_builder/ty2022/states/az/individual_return.rb:12]: This was unused, so I deleted it

File: app/lib/submission_builder/ty2022/states/az/individual_return.rb [hunk 2]
```diff
@@ -4,12 +4,15 @@ module Ty2022
...
+          include DependentRelationshipTable
```
Comment (bengolder) [inline app/lib/submission_builder/ty2022/states/az/individual_return.rb:7]: I put the look up hash into a separate module so that this file wouldn't get so cluttered.

File: app/lib/submission_builder/ty2022/states/az/individual_return.rb [hunk 3]
```diff
@@ -4,12 +4,15 @@ module Ty2022
...
+
+          FILING_STATUS_OPTIONS = {
+            :married_filing_jointly => 'MarriedJoint',
+            :head_of_household => 'HeadHousehold',
+            :married_filing_separately => 'MarriedFilingSeparateReturn',
+            :single => "Single"
+          }
```
Comment (bengolder) [inline app/lib/submission_builder/ty2022/states/az/individual_return.rb:14]: This was already in the file, much lower down. I moved it up next to the other constants.

File: app/models/state_file_dependent.rb
Comment (bengolder) [inline app/models/state_file_dependent.rb:1]: The labels in here are used for all PDFs. I figured that we may want to use these in places other than the PDFs so I included it with the model.

File: app/models/state_file_dependent.rb [hunk 4]
```diff
@@ -47,18 +64,26 @@ def full_name
...
+  end
+
+  def senior?
+    age >= 65
   end
 
   def age
-    ((MultiTenantService.statefile.end_of_current_tax_year.to_time - dob.to_time) / 1.year.seconds).floor
+    MultiTenantService.statefile.current_tax_year - dob.year
+  end
```
Comment (bengolder) [inline app/models/state_file_dependent.rb:84]: I had to change this because it was calculating slightly different ages on servers vs locally. Calculating age using seconds can cause some bugs because years are not the same quantity of seconds. Some years have more seconds than others. Additionally, calling `to_time` can add a timezone to a date that can adjust it to be earlier or later than the original date.

File: app/views/state_file/questions/ny_permanent_address/edit.html.erb [hunk 5]
```diff
@@ -15,7 +15,7 @@
           <div class="form-question spacing-below-15">
             <div class="text--bold"><%=t(".address_header") %></div>
             <div><%= current_intake.direct_file_data.mailing_street %></div>
-            <div><%= current_intake.direct_file_data.mailing_city %>,<%= current_intake.direct_file_data.mailing_state %> <%= current_intake.direct_file_data.mailing_zip %></div>
+            <div><%= current_intake.direct_file_data.mailing_city %>, <%= current_intake.direct_file_data.mailing_state %> <%= current_intake.direct_file_data.mailing_zip %></div>
```
Comment (bengolder) [inline app/views/state_file/questions/ny_permanent_address/edit.html.erb:18]: This is a tiny unrelated fix for how we display addresses

**Review by tofarr — APPROVED** [review]
